src/hotspot/share/compiler/compilerOracle.cpp

```diff
@@ -634,19 +634,24 @@ void skip_comma(char* &line) {
   }
 }
 
-static bool parseEnumValueAsUintx(enum CompileCommand option, const char* line, uintx& value, int& bytes_read, char* errorbuf, const int buf_size) {
+enum parse_result_t { handled_ok, handled_err, not_handled };
+
+static parse_result_t parseEnumValueAsUintx(enum CompileCommand option, const char* line, uintx& value, int& bytes_read, char* errorbuf, const int buf_size) {
   if (option == CompileCommand::MemStat) {
     if (strncasecmp(line, "collect", 7) == 0) {
       value = (uintx)MemStatAction::collect;
+      return handled_ok;
     } else if (strncasecmp(line, "print", 5) == 0) {
       value = (uintx)MemStatAction::print;
       print_final_memstat_report = true;
+      return handled_ok;
     } else {
       jio_snprintf(errorbuf, buf_size, "MemStat: invalid value expected 'collect' or 'print' (omitting value means 'collect')");
+      return handled_err;
```

Comment:
> If "omitting value means 'collect'" then why do we not simply set value = (uintx)MemStatAction::collect ? Otherwise what is that message supposed to mean?

We only enter this function if a value for this command had been given (`-XX:CompileCommand=<command>,<method>[,<value>]`). The default case is handled somewhere else (CompilerOracle::parse_from_line).

> Ternary return values are unpleasant.

As, on a general base? You don't like the number three :-) ?

Here, we have to distinguish between:
- not handled, its not a command option that accepts enum strings
- handled, with two variants: handled, had an error, and handled, had no error. Only in the latter case we want to register the command.

Reply:
Okay I'll butt out. The comment doesn't make sense to me.

Reply:
@dholmes-ora Sorry if I came across as flippant; that was not my intention. If you can stomach it, I rewrote the patch to be hopefully easier to read. Thanks.

Reply:
No I didn't take it that way at all. The new patch does seem easier to read, but I will still defer to compiler folk to review this.